File: app/src/main/java/com/example/csotelo/ashstrolling/core/Point.java
package com.example.csotelo.ashstrolling.core;

public class Point {
    long x, y = 0;

    public Point() {
        x = 0;
        y = 0;
    }

    public Point(long x, long y) {
        this.x = x;
        this.y = y;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return new Point(x, y);
    }

    @Override
    public String toString() {
        return "" + x + "," + y;
    }
}

File: app/src/main/java/com/example/csotelo/ashstrolling/core/Point.kt
package com.example.csotelo.ashstrolling.core

class Point(internal var x: Long = 0, internal var y: Long = 0) {


    @Throws(CloneNotSupportedException::class)
    protected fun clone(): Any {
        return Point(x, y)
    }

    override fun toString(): String {
        return "$x,$y"
    }
}
